### ISSUE TITLE
improve pagination for reports

### ISSUE THREAD

**Issue by bastelfreak** [puppetlabs/puppet-enterprise_issues#30]
## Use Case

In PE 2021 the pagination for reports is a bit challenging when you've many reports:

<img width="346" alt="reports" src="https://github.com/puppetlabs/puppet-enterprise_issues/assets/341866/71f5bc57-116e-4983-84c8-5172531284bc">

I've no option to jump to a specific page or to filter by date. And there are no visible GET parameters I could adjust. I've to click 70 times if I need to go to page 70.

## Describe the Solution You Would Like

Improve the reports view by providing an option to insert a specific page number and/or allow users to filter by date range.

**Comment by kenyon** [comment]
Yes! An option to download the whole report for reading outside the web browser would be good too.

**Comment by steveax** [comment]
Agree that a date-range filter would be very useful.
